Hybrid version

manolo opened the pull request:
Connects to #344 

Don't worry about tests yet. For some reason there are tests failing in travis/sauce that do not fail in local.
This PR is the precursor of the pure 2.0 element port where I'm focusing right now.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-date-picker/366)
<!-- Reviewable:end -->

tomivirkki commented:
We agreed to not focus on P1 tests but P2 must pass on all platforms.

---

Review status: 0 of 24 files reviewed at latest revision, 5 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 16 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9ktxg33tJiehxfI2X:-Kk9ktxg33tJiehxfI2Y:bvpb9ug) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/.travis.yml#L16)):*
> ```YAML
> install:
>   - npm install
>   - polymer install --variants
> ```

Drop the installing of variants here to test P2 only

---

*[bower.json, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9oKES9OS2ZCNhEnvO:-Kk9oKES9OS2ZCNhEnvP:b-3afquz) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/bower.json#L51)):*
> ```JSON
>     "web-component-tester": "^6.0.0"
>   },
>   "variants": {
> ```

I guess there's no need to add the variants block at all if we're not testing it

---

*[vaadin-date-picker-light.html, line 111 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9nwkTf5uGRpST4lVd:-Kk9nwkTf5uGRpST4lVe:b-tyxhjt) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-light.html#L111)):*
> ```HTML
>         // Using the same selector than in combo-box.
>         // TODO: revisit this to decide the selector and document conveniently.
>         return Polymer.dom(this).querySelector('iron-input,input[is="iron-input"],paper-input,.paper-input-input,.input');
> ```

Can drop `input[is="iron-input"`

---

*[vaadin-month-calendar.html, line 234 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9lR3O-5A54Y2_nT_I:-Kk9lR3O-5A54Y2_nT_J:bremfam) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-month-calendar.html#L234)):*
> ```HTML
>       _getWeekDayNames: function(weekDayNames, weekDayNamesShort, showWeekNumbers, firstDayOfWeek) {
>         if (weekDayNames === undefined || weekDayNamesShort === undefined ||
>           showWeekNumbers === undefined || firstDayOfWeek === undefined) return;
> ```

Please use braces as per convention (and as we do in all the other components)

---

*[test/custom-input.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9m8wyEyPTifq6ijIH:-Kk9m8wyEyPTifq6ijII:b1m8t6i) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/test/custom-input.html#L20)):*
> ```HTML
>       <vaadin-date-picker-light>
>         <iron-input>
>           <input is="iron-input">
> ```

Can drop `is="iron-input"` here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-Kk9miEwDkBpOJh2H81u:b-gqu5m5)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-infinite-scroller.html, line 301 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAJoXamxFtsNwyVYhi:-KkAJoXamxFtsNwyVYhj:b2jmgk8) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-infinite-scroller.html#L301)):*
> ```HTML
>       var tmpInstance = itemWrapper.instance;
>       itemWrapper.instance = this.stamp({});
>       Polymer.Element && itemWrapper.instance._flushProperties();
> ```

Apparently `_flushProperties` is renamed to `_enableProperties` in P2 rc9

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Reviewed 2 of 24 files at r1.
Review status: 2 of 24 files reviewed at latest revision, 9 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 22 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAZrrDSkE5u36lNjLi:-KkAZrrDSkE5u36lNjLj:b-97q8nq) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/2c4c45dce3ba1aacf8eba02eb7238311a64efd8b/.travis.yml#L22)):*
> ```YAML
> 
> install:
>   - yarn
> ```

Why is `yarn` dropped?

---

*[vaadin-date-picker-behavior.html, line 696 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAbsyDL4vv2d8NNOwn:-KkAbsyDL4vv2d8NNOwo:b-kvhvpv) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-behavior.html#L696)):*
> ```HTML
> 
>     _userInputValueChanged: function(value) {
>       if (!this.opened && !this._inputValue) {
> ```

Hmm not sure about this check why it has been added? I think we should open the overlay even when, for example,  value is changed to empty with a backspace

---

*[vaadin-date-picker.html, line 214 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkA_Xy6HMlheDuprboN:-KkA_Xy7smniHqdbFkxj:b-8eh1zc) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker.html#L214)):*
> ```HTML
>           placeholder$="[[placeholder]]"
>           aria-label$="[[label]]"
>           invalid="[[invalid]]"
> ```

I think `invalid` is set by the parent `iron-input` ? If not, then `invalid$="[[invalid]]"` is needed

---

*[vaadin-month-calendar.html, line 234 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9lR3O-5A54Y2_nT_I:-KkAcWd905nJAT507LRI:bwy96n6) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-month-calendar.html#L234)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Please use braces as per convention (and as we do in all the other components)
</blockquote></details>

+1 for what Tomi said

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

manolo commented:
This is the hybrid version. As such it needs some things that are removed in pure 2.0. 

---

Review status: 2 of 24 files reviewed at latest revision, 9 unresolved discussions, some commit checks failed.

---

*[.travis.yml, line 16 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9ktxg33tJiehxfI2X:-KkAxzh-lU_PVMGzkk6V:boolrox) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/.travis.yml#L16)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Drop the installing of variants here to test P2 only
</blockquote></details>

This is the hybrid branch so it must have the variants, it is removed in the pure polymer 2.
Removing it here as well, and might enable it if all tests pass eventually for 1.x.

---

*[.travis.yml, line 22 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAZrrDSkE5u36lNjLi:-KkAve8c8zw-0kbKjzz0:bb6jxqi) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/2c4c45dce3ba1aacf8eba02eb7238311a64efd8b/.travis.yml#L22)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Why is `yarn` dropped?
</blockquote></details>

Ups... I just copied these files from an hybrid element. Why do we have mixed installations? if we go with yarn all elements should be updated at the same time.

Done.

---

*[bower.json, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9oKES9OS2ZCNhEnvO:-KkAyaAsFq000yeSevCD:befl7x6) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/bower.json#L51)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

I guess there's no need to add the variants block at all if we're not testing it
</blockquote></details>

This is the Hybrid branch, in pure 2.0 it is removed.

---

*[vaadin-date-picker-light.html, line 111 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9nwkTf5uGRpST4lVd:-KkAyhJDk9wNenQoKHpk:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-light.html#L111)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Can drop `input[is="iron-input"`
</blockquote></details>

Done.

---

*[vaadin-date-picker.html, line 214 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkA_Xy6HMlheDuprboN:-KkAz3Xe9inTAZr3mEdv:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker.html#L214)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

I think `invalid` is set by the parent `iron-input` ? If not, then `invalid$="[[invalid]]"` is needed
</blockquote></details>

Done.

---

*[vaadin-month-calendar.html, line 234 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9lR3O-5A54Y2_nT_I:-KkAtrnBgAi4W5wBetFW:b-yl1p2j) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-month-calendar.html#L234)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

+1 for what Tomi said
</blockquote></details>

I added these lines with a reg-exp, and the linter didn't complain. Anyways it is a very common pattern not embrace return when it's alone, probably because all developers understand perfectly and saves a couple of lines in your ide.

Removed though, I don't have a preference for embracing or not.

---

*[test/custom-input.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9m8wyEyPTifq6ijIH:-KkAuM9qkVI_Py-Y_Lzn:b-qh0s9q) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/test/custom-input.html#L20)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Can drop `is="iron-input"` here
</blockquote></details>

in Hybrid elements it's still needed, I think.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-Kk9v5MXlNQ9Gm8Hnx80:b-725y2m)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
If opened by clicking the calendar, arrow keys scroll the document

---

Review status: 2 of 24 files reviewed at latest revision, 8 unresolved discussions, some commit checks broke.

---

*[bower.json, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9oKES9OS2ZCNhEnvO:-KkB4UGVEalcysmM5eFX:b-p3l23l) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/bower.json#L51)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

This is the Hybrid branch, in pure 2.0 it is removed.
</blockquote></details>

If we leave the variants then we must also test them. How else we can know that they even work? But preferably just remove the whole “variants” block.

---

*[vaadin-infinite-scroller.html, line 301 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAJoXamxFtsNwyVYhi:-KkB6R1WRjMxlVB1vEsb:b-nvddgy) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-infinite-scroller.html#L301)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Apparently `_flushProperties` is renamed to `_enableProperties` in P2 rc9
</blockquote></details>

Or as @platosha discovered, it's not needed at all (if you stamp an empty object)

---

*[vaadin-infinite-scroller.html, line 366 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkB6Kytz0YN3yTh8reA:-KkB6Kyu-1l7bS9MWA40:bcmwg22) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/5c5aec1ad54bd94044717ac4af9524a2d981e7c7/vaadin-infinite-scroller.html#L366)):*
> ```HTML
>         this._buffers.forEach(function(buffer) {
>           [].forEach.call(buffer.children, function(itemWrapper) {
>             // itemWrapper.instance.notifyPath(prop, value, true);
> ```

Why is this commented?

---

*[test/custom-input.html, line 20 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9m8wyEyPTifq6ijIH:-KkB5KZitT0zdVCiVDS-:bx0qahc) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/test/custom-input.html#L20)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

in Hybrid elements it's still needed, I think.
</blockquote></details>

Hybrid `iron-input` doesn't extend.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkB7lzCFEAm2-NUzH2d:b-s32hy9)*
<!-- Sent from Reviewable.io -->

manolo commented:
> If opened by clicking the calendar, arrow keys scroll the document

It seems not happing anymore, or at least I cannot reproduce

---

Review status: 2 of 28 files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---

*[vaadin-date-picker-behavior.html, line 696 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAbsyDL4vv2d8NNOwn:-KkQ6GEJbb744uMaZAtC:b-k1cxta) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-behavior.html#L696)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Hmm not sure about this check why it has been added? I think we should open the overlay even when, for example,  value is changed to empty with a backspace
</blockquote></details>

It makes not sense to parseDate below if it is empty

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkQ6VrRZmKGD_j7ewU4:blav750)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: 2 of 26 files reviewed at latest revision, 9 unresolved discussions.

---

*[.gitignore, line 1 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHJ1KaPKMBDnvXQso:-KkUHJ1KaPKMBDnvXQsp:b-uwxmu1) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.gitignore#L1)):*
> ```
> bower_*
> ```

this should be `bower_components*`

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUHRqDcjDhV0Gr83HD:bgk296g) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
> ```YAML
> 
> install:
>   # not working in travis - yarn
> ```

could you please attach the link to the "not working" build?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

limonte commented:
Label demo doesn't work in P2 (label isn't visible): https://i.imgur.com/5fmHLZm.png

---

Review status: 2 of 26 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkUOpdIM2pYS49Oj4GJ:b1y96sm)*
<!-- Sent from Reviewable.io -->

limonte commented:
The "Custom input" demo doesn't work both in P1 and P2: https://i.imgur.com/yeyFrAj.png

Expected result: When I click any date, the custom input value is set.
Actual result: When I click any date, nothing happens.  

---

Review status: 2 of 26 files reviewed at latest revision, 9 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkUPvsrk0bErG4O-eOS:bokf5pb)*
<!-- Sent from Reviewable.io -->

manolo commented:
> Label demo doesn't work in P2 ...
> The "Custom input" demo doesn't work ...

Fixed we need now to wrap `iron-input` instead of `is=iron-input`, it's a breaking change since I changed the selector because of Tomi's suggestion

---

Review status: 2 of 27 files reviewed at latest revision, 9 unresolved discussions.

---

*[.gitignore, line 1 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHJ1KaPKMBDnvXQso:-KkUSOomimSC-YIcaiwV:by9oqil) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.gitignore#L1)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

this should be `bower_components*`
</blockquote></details>

It covers both and it's aligned with the  `bower-*` in the next line, actually we only want `bower.json`.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUSyb55rL58nHDFiXT:b-biqcn3) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

could you please attach the link to the "not working" build?
</blockquote></details>

Build failing:
https://travis-ci.org/vaadin/vaadin-date-picker/builds/232456694

A thread with the same problem:
https://github.com/Polymer/polymer-cli/issues/666

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkUT6CIuf4xaN4CayK2:by09ko1)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: 2 of 27 files reviewed at latest revision, 9 unresolved discussions, some commit checks broke.

---

*[.gitignore, line 1 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHJ1KaPKMBDnvXQso:-KkUXubVnQvdaBYjjDYF:b-gtv6ii) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.gitignore#L1)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

It covers both and it's aligned with the  `bower-*` in the next line, actually we only want `bower.json`.
</blockquote></details>

https://github.com/PolymerElements/paper-input/blob/master/.gitignore

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

limonte commented:
Labels are still invisible for all demos 

---

Review status: 2 of 27 files reviewed at latest revision, 9 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUZlnHplrTkjPmkB3w:bp5vcuv) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Build failing:
https://travis-ci.org/vaadin/vaadin-date-picker/builds/232456694

A thread with the same problem:
https://github.com/Polymer/polymer-cli/issues/666
</blockquote></details>

That's because of introducing `run`. I don't like this configuration, why do you need to specify filenames in Travis config?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkU_GUPkFRaOCTNu95v:b-hebgfb)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 3 of 25 files at r1, 2 of 5 files at r2, 5 of 16 files at r3, 19 of 19 files at r4, 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---

*[vaadin-date-picker.html, line 82 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUYqyWiRjxT2wFOhjq:-KkUYqyWiRjxT2wFOhjr:b-16pwe6) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/vaadin-date-picker.html#L82)):*
> ```HTML
>       }
> 
>       [name=suffix] {
> ```

Can we use `#suffix` selector and  `id="suffix"` on the element?

---

*[vaadin-infinite-scroller.html, line 179 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUZsnRZkzDJb9uF-Fe:-KkUZsnRZkzDJb9uF-Ff:bk43jka) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/vaadin-infinite-scroller.html#L179)):*
> ```HTML
>       this.transform('translate3d(0, ' + this._buffers[index].translateY + 'px, 0)', this._buffers[index]);
>       this._buffers[index].updated = false;
> ```

Unnecessary whitespace change.

---

*[test/basic.html, line 52 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkU_uUwDGovJTsXeu5C:-KkU_uUxZy5f2XjfNp_d:b-hd3air) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/basic.html#L52)):*
> ```HTML
>         datepicker.value = '2000-02-01';
> 
>         // IE also needs timeouts
> ```

Strange.

Maybe try `flush(callback);` or `Polymer.dom.flush(); Polymer.updateStyles();` instead of the timeout?

---

*[test/keyboard-input.html, line 291 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUbK-ADVVdomL1WDW6:-KkUbK-ADVVdomL1WDW7:bu087m7) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/keyboard-input.html#L291)):*
> ```HTML
>             });
>             var spy = sinon.spy(e, 'preventDefault');
>             datepicker.dispatchEvent(e);
> ```

Does not look correct. It was testing events from the native input, now from the wrapper datepicker.

---

*[test/keyboard-navigation.html, line 305 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUbshufXI0g8w2H7nT:-KkUbshv6rAz4Px9dwnD:bx4gn3z) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/keyboard-navigation.html#L305)):*
> ```HTML
> 
>             setTimeout(function() {
>               var delta = Math.abs(overlay.$.scroller.position - initialPosition);
> ```

Use expect with delta then.

`expect(overlay.$.scroller.position).to.be.closeTo(initialPosition, 1);`

This would produce some more meaning if failed.

---

*[test/late-upgrade.html, line 36 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUce0drzP01MNrZYLf:-KkUce0drzP01MNrZYLg:b-678it2) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/late-upgrade.html#L36)):*
> ```HTML
>         // Assure that the elements has not been upgraded yet
>         var is = Polymer.Element ? datepicker instanceof Polymer.Element : Polymer.isInstance(datepicker);
>         expect(is).to.be.false;
> ```

`expect(datepicker.is).to.be.undefined`?

---

*[test/overlay.html, line 251 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUd4yS8BTe3_vrXzER:-KkUd4yS8BTe3_vrXzES:b-syatco) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/overlay.html#L251)):*
> ```HTML
> 
>               expect(overlay.$.scroller.position).to.equal(lastScrollPos);
>               // FIXME: fails in FF + Polymer 1.x
> ```

Maybe fine to uncomment, since not testing Polymer 1.x

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUfSaGVeMWG4I-Cekg:beecbt8) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

That's because of introducing `run`. I don't like this configuration, why do you need to specify filenames in Travis config?
</blockquote></details>

Here's the same approach for the grid with yarn working: https://github.com/vaadin/vaadin-grid/blob/master/.travis.yml

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---

*[.gitignore, line 1 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHJ1KaPKMBDnvXQso:-KkUfWdfUFyh4mq-pr1E:b-yjbkg0) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.gitignore#L1)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

https://github.com/PolymerElements/paper-input/blob/master/.gitignore
</blockquote></details>

Does not convince me a matter of taste again. Is this a blocker ?

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUf3icYzedzo7-Sk9G:bnabtaj) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

That's because of introducing `run`. I don't like this configuration, why do you need to specify filenames in Travis config?
</blockquote></details>

I don't follow what happens with run ? in the failed build there wasn't the `$run` but `polymer test` lines, `$run` was introduced in  a latest commit for simplicity.

About specifying filenames in tests is to split executions and bypassing the problem of memory leaks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

limonte commented:
Review status: all files reviewed at latest revision, 16 unresolved discussions, some commit checks broke.

---

*[.gitignore, line 1 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHJ1KaPKMBDnvXQso:-KkUhEmUjAzCgWCts-HP:b-t6rd13) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.gitignore#L1)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Does not convince me a matter of taste again. Is this a blocker ?
</blockquote></details>

not a blocker

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUhNbkYv3gJycWRdVe:bj8pelo) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I don't follow what happens with run ? in the failed build there wasn't the `$run` but `polymer test` lines, `$run` was introduced in  a latest commit for simplicity.

About specifying filenames in tests is to split executions and bypassing the problem of memory leaks.


</blockquote></details>

Take a look on https://github.com/vaadin/vaadin-grid/blob/master/.travis.yml - yarn is working there fine

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
> It seems not happing anymore, or at least I cannot reproduce

:+1: Seems to be fixed now.

Discovered one issue. On mobile fullscreen mode the selected date is not reflected to the top bar. This is how it should look like: https://www.dropbox.com/s/v3adoyugwzzp407/Screenshot%202017-05-19%2011.26.47.png?dl=0

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkUmgOvA0y3r2u1Tytk:b7uzk6w)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks broke.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkUrMSuO58SpEOavASA:b-4u00vj) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

Take a look on https://github.com/vaadin/vaadin-grid/blob/master/.travis.yml - yarn is working there fine
</blockquote></details>

Can you try the same with date-picker ? it could be a different dependencies or something, I'm pretty sure `$run` has nothing to do.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

manolo commented:
> On mobile fullscreen mode the selected date is not reflected to the top bar

missing the `slot=input` attribute. Done

---

Review status: 22 of 27 files reviewed at latest revision, 15 unresolved discussions.

---

*[vaadin-date-picker.html, line 82 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUYqyWiRjxT2wFOhjq:-KkUroDQc0g8xmFnFKCA:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/vaadin-date-picker.html#L82)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Can we use `#suffix` selector and  `id="suffix"` on the element?
</blockquote></details>

Done.

---

*[vaadin-infinite-scroller.html, line 179 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUZsnRZkzDJb9uF-Fe:-KkUs4Eoy4OBZLhiil8P:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/vaadin-infinite-scroller.html#L179)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Unnecessary whitespace change.
</blockquote></details>

Done.

---

*[test/basic.html, line 52 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkU_uUwDGovJTsXeu5C:-KkUs7w4SG7ai_4hxETM:bembcgx) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/basic.html#L52)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Strange.

Maybe try `flush(callback);` or `Polymer.dom.flush(); Polymer.updateStyles();` instead of the timeout?
</blockquote></details>

Since with this timeout it works, I do prefer to spend time on this flaky test in the Pure the polymer 2 PR.  Is it a blocker ?

---

*[test/keyboard-input.html, line 291 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUbK-ADVVdomL1WDW6:-KkUxsY17fF-_d8TK0zL:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/keyboard-input.html#L291)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Does not look correct. It was testing events from the native input, now from the wrapper datepicker.
</blockquote></details>

Done.

---

*[test/keyboard-navigation.html, line 305 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUbshufXI0g8w2H7nT:-KkUxuHtWGiRHQZ5GSgD:b-896fix) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/keyboard-navigation.html#L305)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Use expect with delta then.

`expect(overlay.$.scroller.position).to.be.closeTo(initialPosition, 1);`

This would produce some more meaning if failed.
</blockquote></details>

Done.

---

*[test/late-upgrade.html, line 36 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUce0drzP01MNrZYLf:-KkUz1RBHFXsq_0FXNu3:b-s9a24) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/late-upgrade.html#L36)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

`expect(datepicker.is).to.be.undefined`?
</blockquote></details>

For some reason using `.is` was causing a  `Error: Maximum call stack size exceeded` in Chrome, so I ended with this solution.
But anyways is test code, I don't think it's a blocker.

---

*[test/overlay.html, line 251 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUd4yS8BTe3_vrXzER:-KkUzPGqKie0FdEZWZFQ:b8zfaot) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/overlay.html#L251)):*
<details><summary><i>Previously, platosha (Anton Platonov) wrote…</i></summary><blockquote>

Maybe fine to uncomment, since not testing Polymer 1.x
</blockquote></details>

Not possible since I'm  testing in both polymer 1 and 2, see travis conf : `polymer install --variants` 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkV1lPJSCjrLo9HvWS2:b5gz38x)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 22 of 27 files reviewed at latest revision, 15 unresolved discussions.

---

*[vaadin-infinite-scroller.html, line 366 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkB6Kytz0YN3yTh8reA:-KkViKCEaKjaKkVBri97:b77r9v3) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/5c5aec1ad54bd94044717ac4af9524a2d981e7c7/vaadin-infinite-scroller.html#L366)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Apparently is not needed. Just left a comment because I need to confirm with @saulis 
</blockquote></details>

Asked him, and decided to remove, Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 5 of 5 files at r6.
Review status: all files reviewed at latest revision, 10 unresolved discussions.

---

*[test/basic.html, line 52 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkU_uUwDGovJTsXeu5C:-KkVyE41wxV_sNBNyH5d:b-o07bwx) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/basic.html#L52)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Since with this timeout it works, I do prefer to spend time on this flaky test in the Pure the polymer 2 PR.  Is it a blocker ?
</blockquote></details>

OK not a blocker

---

*[test/late-upgrade.html, line 36 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUce0drzP01MNrZYLf:-KkVyPzZhtAp45IbXEnf:b32w0x) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/late-upgrade.html#L36)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

For some reason using `.is` was causing a  `Error: Maximum call stack size exceeded` in Chrome, so I ended with this solution.
But anyways is test code, I don't think it's a blocker.
</blockquote></details>

OK

---

*[test/overlay.html, line 251 at r5](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUd4yS8BTe3_vrXzER:-KkVyT1_4AE3-qJGMEq-:b32w0x) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/dd93df050c776f7868c3c3a75498cd970aaa7625/test/overlay.html#L251)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Not possible since I'm  testing in both polymer 1 and 2, see travis conf : `polymer install --variants` 
</blockquote></details>

OK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

limonte commented:
Labels are still invisible for all demos for P2

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-Kkj35Ga4IfL99xqtOjS:biwl4l1) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Can you try the same with date-picker ? it could be a different dependencies or something, I'm pretty sure `$run` has nothing to do.
</blockquote></details>

All right, let's skip this for now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-Kkj3IbJHT21hy4xBstf:b-ag7g6g)*
<!-- Sent from Reviewable.io -->

manolo commented:
> Labels are still invisible for all demos for P2

Good catch, slot attribute was not set. Done.

---

Reviewed 1 of 5 files at r6.
Review status: 24 of 27 files reviewed at latest revision, 7 unresolved discussions.

---

*[.travis.yml, line 20 at r4](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkUHRqDcjDhV0Gr83HC:-KkjIDNmgruvPRBB-629:b-sybj35) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/aa27dce64e0bb0dbffe234ec9e1b0dae685a4aaf/.travis.yml#L20)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

All right, let's skip this for now.
</blockquote></details>

Working after cleaning the cache. We need to remember to clean it when merging to master. 
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkjI684m0ZcgbW_Snqt:bkf61gm)*
<!-- Sent from Reviewable.io -->

limonte commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> :+1: :tada: :cake: 

---

Reviewed 3 of 25 files at r1, 2 of 5 files at r2, 2 of 16 files at r3, 13 of 19 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-KkjJFeWudRMDEOKvNDQ:b-u92d6j)*
<!-- Sent from Reviewable.io -->

Saulis commented:
Review status: 26 of 27 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-date-picker-behavior.html, line 696 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAbsyDL4vv2d8NNOwn:-Kkj_5PxZK7JvkDxvBMf:bvxsrsn) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-behavior.html#L696)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

It makes not sense to parseDate below if it is empty
</blockquote></details>

Well no, but it should still open the overlay with empty input like it has in original implementation.

---

*[vaadin-infinite-scroller.html, line 366 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkB6Kytz0YN3yTh8reA:-Kkja2S_5xKZfzDLLYXV:biq555w) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/5c5aec1ad54bd94044717ac4af9524a2d981e7c7/vaadin-infinite-scroller.html#L366)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

Asked him, and decided to remove, Done.
</blockquote></details>

Yeah, basically this function is needed to update parent property paths – currently `i18n` is the only deep property which is propagated from parent to overlay. So, in practice, this is needed to for updating something like `picker.set('i18n.monthNames.0, 'foobar')` on-the-fly. 

However, updating `i18n.x` properties on-the-fly haven't worked so far since we don't have any `i18n.*` observers in place in the template, so I advised to remove this functionality for now and adding it back later if somebody actually needs fully dynamic `i18n` properties.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[bower.json, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9oKES9OS2ZCNhEnvO:-Kkjg0qkKcCQkI9OpgvE:b-74knlw) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/bower.json#L51)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary><blockquote>

I'll leave the variants here until the PR while I'm working with it. I'll remove them when moved to review columnt.
</blockquote></details>

The variant is still here even though the PR is in review

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

platosha commented:
Reviewed 1 of 1 files at r8.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 25 of 27 files reviewed at latest revision, 2 unresolved discussions.

---

*[bower.json, line 51 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-Kk9oKES9OS2ZCNhEnvO:-KkjgvVsLXFqo7QTq2PL:b-tqofpy) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/bower.json#L51)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

The variant is still here even though the PR is in review
</blockquote></details>

Finally, variants must be here since this PR is for the hybrid version.

---

*[vaadin-date-picker-behavior.html, line 696 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-KkAbsyDL4vv2d8NNOwn:-Kkk419GeP3N_cn6uc74:bb6un1j) ([raw file](https://github.com/vaadin/vaadin-date-picker/blob/189121883e8f2aef108743aca9402a32870a51c3/vaadin-date-picker-behavior.html#L696)):*
<details><summary><i>Previously, Saulis (Sauli Tähkäpää) wrote…</i></summary><blockquote>

Well no, but it should still open the overlay with empty input like it has in original implementation.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366)*
<!-- Sent from Reviewable.io -->

Saulis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> after build passes

---

Reviewed 1 of 25 files at r1, 2 of 5 files at r2, 2 of 16 files at r3, 13 of 19 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 2 of 3 files at r7, 1 of 1 files at r8.
Review status: 25 of 27 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-Kkk54KZj-iTgpGJCiWD:b-9pmja4)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 25 files at r1, 2 of 5 files at r2, 2 of 16 files at r3, 11 of 19 files at r4, 2 of 2 files at r5, 3 of 5 files at r6, 2 of 3 files at r7, 1 of 1 files at r8, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-date-picker/366#-:-Kkjh6nyNF1ElXYIuZPb:bnfp4nl)*
<!-- Sent from Reviewable.io -->
